lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -6300,8 +6300,23 @@ struct ExportVerilogPass : public ExportVerilogBase<ExportVerilogPass> {
 private:
   raw_ostream &os;
 };
+
+struct ExportVerilogStreamOwnedPass : public ExportVerilogPass {
+  ExportVerilogStreamOwnedPass(std::unique_ptr<llvm::raw_ostream> os)
+      : ExportVerilogPass{*os} {
+    owned = std::move(os);
+  }
+
+private:
+  std::unique_ptr<llvm::raw_ostream> owned;
+};
```

Comment:
External programs will run the pass like this way:

```c
addPass(pm, /* ... */);
addPass(pm, ExportVerilogPass, callback);
addPass(pm, /* ... */);

runPasses(pm);
```

The callback will not be called in the `addPass` function, so it needs to be stored somewhere, that's the reason why I added the new owned-version class.